src/components/planner/sidebar/sessionController/Refresh.tsx

```diff
@@ -30,31 +32,60 @@ const Refresh = () => {
           const updatedClasses = await BackendAPI.getCourseClass(course);
           return {
             ...course,
-            hash: mismatchedMap.get(course.id) as string,
+            hash: mismatchedMap.get(course.id),
             classes: updatedClasses,
           };
         })
       );
+
       const finalCourses = pickedCourses.map((course) =>
         mismatchedMap.has(course.id)
           ? updatedCoursesWithClasses.find(
               (updated) => updated.id === course.id
             ) || course
           : course
       );
+
+      const classIdMap: Record<number, number> = {};
+      pickedCourses.forEach((course) => {
+        const oldClasses = course.classes || [];
+        const newClasses = finalCourses.find((finalCourse) => finalCourse.id === course.id)?.classes || [];
+        oldClasses.forEach((oldClass) => {
+          const newClass = newClasses.find((cls) => cls.name === oldClass.name);
+          if (newClass) {
+            classIdMap[oldClass.id] = newClass.id;
+          }else{
+            classIdMap[oldClass.id] = null;
+          }
+        });
+      });
+
+      let updatedMultipleOptions = [...multipleOptions];
+      for (let i = 0; i < updatedMultipleOptions.length; i++) {
+        for (let j = 0; j < updatedMultipleOptions[i].course_options.length; j++) {
+          const courseOption = updatedMultipleOptions[i].course_options[j];
+          const oldClassId = courseOption.picked_class_id;
+          if (classIdMap[oldClassId] !== undefined ) {
```

Comment:
Maybe it would be best to just have `if(classIdMap[oldClassId]) { ... }` without the `!== undefined`.

Reply:
The reason for that is that when a class for instance 1LEIC01 does no longer exist I create a null entry in the map (ex : 1138: null) therefore there in that portion of the code  null is dif from undefined so it sets the value of picked class id to null if it was just if(classIdMap[oldClassId]) { ... } it would not change the value and the picked class id would remain the previous one and it would crash because It would not find a class with the picked id( this worked before because of the if statement that I removed now)